src/frontend/js/widgets/Dashboard/hooks/useTeacherDashboardRouter/index.tsx

```diff
@@ -0,0 +1,168 @@
+import { useIntl } from 'react-intl';
+import { Navigate, RouteObject } from 'react-router-dom';
+import { TeacherProfileSettingsDashboardLoader } from 'pages/TeacherProfileSettingsDashboardLoader';
+import { TeacherUniversityCourseDashboardLoader } from 'pages/TeacherUniversityCourseDashboardLoader';
+import RouteInfo from 'widgets/Dashboard/components/RouteInfo';
+import { TeacherProfileDashboardLoader } from 'widgets/Dashboard/components/TeacherProfileDashboardLoader';
+import NavigateWithParams from 'widgets/Dashboard/components/NavigateWithParams';
+import { getDashboardRouteLabel, getDashboardRoutePath } from 'widgets/Dashboard/utils/routers';
+import {
+  TeacherDashboardPaths,
+  teacherDashboardRouteLabels,
+} from 'widgets/Dashboard/utils/teacherRouter';
+
+export function getTeacherDashboardRoutes() {
+  const intl = useIntl();
+  const getRoutePath = getDashboardRoutePath(intl);
+  const getRouteLabel = getDashboardRouteLabel(intl);
+  const routes: RouteObject = {
+    path: getRoutePath(TeacherDashboardPaths.ROOT),
+    handle: {
+      renderLayout: true,
+    },
+    children: [
+      {
+        index: true,
+        element: <Navigate to={getRoutePath(TeacherDashboardPaths.TEACHER_COURSES)} replace />,
+      },
+      {
+        path: getRoutePath(TeacherDashboardPaths.TEACHER_PROFILE),
+        children: [
+          {
+            index: true,
+            element: <Navigate to={getRoutePath(TeacherDashboardPaths.TEACHER_COURSES)} replace />,
+          },
+          {
+            path: getRoutePath(TeacherDashboardPaths.TEACHER_COURSES),
+            handle: {
+              crumbLabel: teacherDashboardRouteLabels[TeacherDashboardPaths.TEACHER_COURSES],
+            },
+            element: <TeacherProfileDashboardLoader />,
+          },
+          {
+            path: getRoutePath(TeacherDashboardPaths.TEACHER_SETTING),
+            handle: {
+              crumbLabel: teacherDashboardRouteLabels[TeacherDashboardPaths.TEACHER_SETTING],
+            },
+            element: <TeacherProfileSettingsDashboardLoader />,
+          },
+          {
+            path: getRoutePath(TeacherDashboardPaths.TEACHER_NOTIFICATIONS),
+            handle: {
+              crumbLabel: teacherDashboardRouteLabels[TeacherDashboardPaths.TEACHER_NOTIFICATIONS],
+            },
+            element: (
+              <RouteInfo title={getRouteLabel(TeacherDashboardPaths.TEACHER_NOTIFICATIONS)} />
+            ),
+          },
+        ],
+      },
+      {
+        path: getRoutePath(TeacherDashboardPaths.UNIVERSITY, {
+          universityId: ':universityId',
```

Comment:
IMO ( to be confirmed by @jbpenrath ) the term `university` should not be used, but instead `organization` which is more generic. Richie could be used by other entities than universities.